File: Views_NodeJS/apis/common.api.js
const constants = require('../constants/index.constant');
const { axiosJava, axiosCSharp } = require('./axiosClient');

const BASE_URL = '/common';

const commonApi = {
  getAllProvince: () => {
    return axiosJava.get(`${BASE_URL}/province-all`);
  },

  getDistrict: (provinceId) => {
    return axiosJava.get(`${BASE_URL}/district/${provinceId}`);
  },

  getWard: (districtId) => {
    return axiosJava.get(`${BASE_URL}/ward/${districtId}`);
  },

  getDistrictById: (districtId) => {
    return axiosJava.get(`${BASE_URL}/district?districtId=${districtId}`);
  },

  getWardById: (wardId) => {
    return axiosJava.get(`${BASE_URL}/ward?wardId=${wardId}`);
  },

  getProvinceById: (provinceId) => {
    return axiosJava.get(`${BASE_URL}/province?provinceId=${provinceId}`);
  },

  getAllPayment: () => {
    return axiosJava.get(`${BASE_URL}/year-payment`);
  },

  getProductEachType: (group) => {
    return axiosCSharp.get(`${BASE_URL}/type/amount-type?group=${group}`);
  },

  getProductsByGroupType: (
    grouptype = 1,
    page = 1,
    pageSize = constants.PAGE_SIZE
  ) => {
    return axiosCSharp.get(
      `${BASE_URL}/products-by-grouptype/${grouptype}?page=${page}&pageSize=${pageSize}`
    );
  },

  getDetailProduct: (productId) => {
    return axiosCSharp.get(`${BASE_URL}/product?productId=${productId}`);
  },

  getImagesOfProduct: (productId) => {
    return axiosCSharp.get(`${BASE_URL}/product/images?productId=${productId}`);
  },

  getProductSold: (productId) => {
    return axiosCSharp.get(`${BASE_URL}/product/sold?productId=${productId}`);
  },

  getProductFeedback: (productId) => {
    return axiosCSharp.get(
      `${BASE_URL}/product/feedback?productId=${productId}`
    );
  },

  getProductStore: (productId) => {
    return axiosCSharp.get(`${BASE_URL}/store/info?productId=${productId}`);
  },

  getProductType: (typeId) => {
    return axiosCSharp.get(`${BASE_URL}/product/type?typeId=${typeId}`);
  },

  getTypeByGroup: (groupId) => {
    return axiosCSharp.get(
      `${BASE_URL}/product/type-by-group?groupId=${groupId}`
    );
  },

  getProductByType: (typeId, page, pageSize) => {
    return axiosCSharp.get(
      `${BASE_URL}/product/product-by-type?typeId=${typeId}&page=${page}&pageSize=${pageSize}`
    );
  },

  getProductsBySeach: (keyword, page, pageSize) => {
    return axiosCSharp.get(
      `${BASE_URL}/product/search?keyword=${keyword}&page=${page}&pageSize=${pageSize}`
    );
  },

  getStoreInfo: (storeId) => {
    return axiosCSharp.get(`${BASE_URL}/store/info/${storeId}`);
  },

  getProductForCart: (productId) => {
    return axiosCSharp.get(`${BASE_URL}/product-for-cart?productId=${productId}`);
  }
};

module.exports = commonApi;

File: API_JAVASpringBoot/API_JAVA/src/main/java/api/java/apis/CommonApi.java
package api.java.apis;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import api.java.entities.District;
import api.java.entities.Province;
import api.java.entities.Ward;
import api.java.entities.Payment;
import api.java.services.CommonService;

@RestController
@RequestMapping(path = "/api/common")
public class CommonApi {
    @Autowired
    private CommonService commonService;

    @GetMapping(path = "/province-all")
    public List<Province> getAllProvince() {
        return commonService.getAllProvinces();
    }

    @GetMapping(path = "/district/{provinceId}")
    public List<District> getDistrict(@PathVariable int provinceId) {
        return commonService.getDistrict(provinceId);
    }

    @GetMapping(path = "/ward/{districtId}")
    public List<Ward> getWard(@PathVariable int districtId) {
        return commonService.getWard(districtId);
    }

    @GetMapping(path = "/province")
    public Province getProvinceById(@RequestParam int provinceId) {
        return commonService.getProvinceById(provinceId);
    }

    @GetMapping(path = "/district")
    public District getDistrictById(@RequestParam int districtId) {
        return commonService.getDistrictById(districtId);
    }

    @GetMapping(path = "/ward")
    public Ward getWardById(@RequestParam int wardId) {
        return commonService.getWardById(wardId);
    }

    @GetMapping(path = "/payment")
    public List<Payment> getAllYearPayment() {
        return commonService.getAllPayments();

    }
}